packages/backend/server/src/app.module.ts

```diff
@@ -225,40 +159,33 @@ export function buildAppModule() {
       NotificationModule,
       MailModule
     )
-
+    // renderer server only
+    .useIf(() => env.flavors.renderer, DocRendererModule)
     // sync server only
-    .useIf(config => config.flavor.sync, SyncModule)
-
+    .useIf(() => env.flavors.sync, SyncModule)
     // graphql server only
     .useIf(
-      config => config.flavor.graphql,
-      VersionModule,
+      () => env.flavors.graphql,
       GqlModule,
+      VersionModule,
       StorageModule,
       ServerConfigModule,
       WorkspaceModule,
-      LicenseModule
+      LicenseModule,
+      PaymentModule,
+      CopilotModule,
+      CaptchaModule,
+      OAuthModule
     )
-
     // doc service only
-    .useIf(config => config.flavor.doc, DocServiceModule)
-
+    .useIf(() => env.flavors.doc, DocServiceModule)
     // self hosted server only
-    .useIf(config => config.isSelfhosted, SelfhostModule)
-    .useIf(config => config.flavor.renderer, DocRendererModule);
-
-  // plugin modules
-  ENABLED_PLUGINS.forEach(name => {
-    const plugin = REGISTERED_PLUGINS.get(name);
-    if (!plugin) {
-      new Logger('AppBuilder').warn(`Unknown plugin ${name}`);
-      return;
-    }
+    .useIf(() => env.selfhosted, WorkerModule, SelfhostModule)
 
-    factor.use(plugin);
-  });
+    // gcloud
+    .useIf(() => env.gcp, GCloudModule);
```

Comment:
add gcp env in deploy

Reply:
Will self-host deployment on GCP also be forced to open? The `PLATFORM` environment variable is set by the user, right? The GCP environment won't set it automatically?

Reply:
it won't